common/static/common/js/spec/components/paging_collection_spec.js

```diff
@@ -0,0 +1,238 @@
+            });
+
+            it('can set the filter field and get the display name', function () {
+                collection.registerFilterableField('test_field', 'Test Field');
+                collection.setFilterField('test_field');
+                expect(requests.length).toBe(1);
+                // The default implementation does not send any query params for filtering
+                expect(collection.filterField).toBe('test_field');
+                expect(collection.filterDisplayName()).toBe('Test Field');
+            });
+
+            it('can set the sort direction', function () {
+                collection.setSortDirection(PagingCollection.SortDirection.ASCENDING);
+                expect(requests.length).toBe(1);
+                // The default implementation does not send any query params for sort direction
+                expect(collection.sortDirection).toBe(PagingCollection.SortDirection.ASCENDING);
+                collection.setSortDirection(PagingCollection.SortDirection.DESCENDING);
+                expect(requests.length).toBe(2);
+                expect(collection.sortDirection).toBe(PagingCollection.SortDirection.DESCENDING);
+            });
+
+            it('can toggle the sort direction when setting the sort field', function () {
+                collection.registerSortableField('test_field', 'Test Field');
+                collection.registerSortableField('test_field_2', 'Test Field 2');
+                collection.setSortField('test_field', true);
+                expect(collection.sortDirection).toBe(PagingCollection.SortDirection.DESCENDING);
+                collection.setSortField('test_field', true);
+                expect(collection.sortDirection).toBe(PagingCollection.SortDirection.ASCENDING);
+                collection.setSortField('test_field', true);
+                expect(collection.sortDirection).toBe(PagingCollection.SortDirection.DESCENDING);
+                collection.setSortField('test_field_2', true);
+                expect(collection.sortDirection).toBe(PagingCollection.SortDirection.DESCENDING);
+            });
+
+            SpecHelpers.withData({
+                'queries with page, page_size, and sort_order parameters when zero indexed': [true, 2],
+                'queries with page, page_size, and sort_order parameters when one indexed': [false, 3],
+            }, function (isZeroIndexed, page) {
+                collection.isZeroIndexed = isZeroIndexed;
+                collection.perPage = 5;
+                collection.sortField = 'test_field';
+                collection.setPage(3);
+                assertQueryParams({'page': page.toString(), 'page_size': '5', 'sort_order': 'test_field'});
+            });
+
+            SpecHelpers.withConfiguration({
+                'using a zero indexed collection': [true],
+                'using a one indexed collection': [false]
+            }, function (isZeroIndexed) {
```

Comment:
I'm conflicted as to whether this would be clearer as just a single function that took the configuration parameters (as DDT does). That may just be too subtle.

Also, have you seen the various plugins that add data-driven testing to Jasmine?

https://github.com/jasmine/jasmine/issues/480